sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -207,3 +207,20 @@ def generate_body():
         (1 - sqrt(3)/3)*N.x + sqrt(6)*sin(theta + pi/4)/3*N.y - sqrt(6)*cos(theta + pi/4)/3*N.z
     assert C.masscenter.vel(N).express(N).simplify() == sqrt(6)*omega*cos(theta + pi/4)/3*N.y + sqrt(6)*omega*sin(theta + pi/4)/3*N.z
     assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    a, b, c = symbols('a b c')
+    m, n = symbols('m n')
+    PinJoint('J', P, C, parent_joint_pos=m*N.x, child_joint_pos=n*A.x, child_axis=a*A.x+b*A.y-c*A.z, parent_axis=a*N.x-b*N.y+c*N.z)
+    assert (a*N.x-b*N.y+c*N.z).angle_between(a*A.x-b*A.y+c*A.z).simplify() == 0 #Axis are aligned
+    assert (a*A.x-b*A.y+c*A.z).express(N).simplify() == sqrt(3)*N.x
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y - omega*A.z)/sqrt(3)
+    assert A.ang_vel_in(N).magnitude() == omega
+    assert A.ang_vel_in(N).angle_between(a*A.x-b*A.y+c*A.z).simplify() == 0
+    assert C.masscenter.vel(N).simplify() == (omega * A.y + omega * A.z)/sqrt(3)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
```

Comment:
These should nave m and n too.

Reply:
Actually these tests were taking forever to run on my system , so I'm trying to use github actions, these tests are work in progress

Reply:
Ok. depending on the complexity of expressions, simplify() may not be appropriate for the unit tests.

I don't think these tests should reach that complexity though. It usually takes several pin joints in succession to make the equations complex enough that simplify() takes too long.

Reply:
Also, you should run the tests locally vs setting up github actions.

Reply:
They are running locally for past 3hours and still running.

Reply:
Then there is a problem with the algorithm we are using. Can you profile the code and find out what is hanging?

Reply:
This tool is good for things that hang: https://github.com/joerick/pyinstrument

Reply:
> Then there is a problem with the algorithm we are using. 

I think it's something related to use of `symbols` and `simplify` in axes as it is making problem really very complicated and lengthy.  Same problem with similar axes just without axes without symbols(unit in every direction) works fine , so the algo is fine I believe.

Reply:
A rule of thumb for you is that you should rare, if ever, be using simplify in the library code and examples. It should only be used in examples or unit tests to simplify the low complexity expressions. Best to write all of your tests and examples without simplify.

Reply:
>  It should only be used in examples or unit tests to simplify the low complexity expressions.

Yes in current tests, simplify is just solving the angle and simplify the velocity, because these expression were simple but just unsolved, so solving them.

Reply:
Simplify just calls many other simplification functions. It's much better to choose a specific simplification function that actually does what you want like `together`, `cancel`, `expand`, `ratsimp` etc. When you call `simplify` it basically just tries all of these functions.

Reply:
Yes, generally trigsimp is most helpful in the mechanics package.

Reply:
But I think for this case we are just solving the result returned, so `simplify` seems fine to me.

Reply:
I don't understand what you mean. Solving and simplifying are not the same thing.

What is the expression before calling simplify?

Reply:
> What is the expression before calling simplify?

```
>>> N.x.angle_between(A.x + A.y)
acos(sqrt(2)*(sqrt(2)*(1/2 - cos(J_theta(t))/2)/2 + sqrt(2)*(cos(J_theta(t))/2 + 1/2)/2))

>>> N.x.angle_between(A.x + A.y).simplify()
0
```

Reply:
```python
In [9]: e = acos(sqrt(2)*(sqrt(2)*(S(1)/2 - cos(J_theta(t))/2)/2 + sqrt(2)*(cos(J_theta(t))/2 + S(1)/2)/2))                                                   

In [10]: e                                                                                                                                                    
Out[10]: 
    ⎛   ⎛   ⎛1   cos(Jₜₕₑₜₐ(t))⎞      ⎛cos(Jₜₕₑₜₐ(t))   1⎞⎞⎞
    ⎜   ⎜√2⋅⎜─ - ──────────────⎟   √2⋅⎜────────────── + ─⎟⎟⎟
    ⎜   ⎜   ⎝2         2       ⎠      ⎝      2          2⎠⎟⎟
acos⎜√2⋅⎜─────────────────────── + ───────────────────────⎟⎟
    ⎝   ⎝           2                         2           ⎠⎠

In [11]: expand_mul(e)                                                                                                                                        
Out[11]: 0
```
Compare timings:
```python
In [1]: J_theta = Function('J_theta')                                                                                                                         

In [2]: e = acos(sqrt(2)*(sqrt(2)*(S(1)/2 - cos(J_theta(t))/2)/2 + sqrt(2)*(cos(J_theta(t))/2 + S(1)/2)/2))                                                   

In [3]: %time expand_mul(e)                                                                                                                                   
CPU times: user 2.26 ms, sys: 22 µs, total: 2.29 ms
Wall time: 2.33 ms
Out[3]: 0

In [4]: %time simplify(e)                                                                                                                                     
CPU times: user 301 ms, sys: 7.13 ms, total: 309 ms
Wall time: 308 ms
Out[4]: 0
```
Calling simplify here wastes 1/3 of a second. In the context of unit tests where we know exactly what expression we are dealing with there is very rarely a good reason to use `simplify`. We need to keep the test suite as fast as possible (it is already way too slow).

Reply:
@moorepants @oscarbenjamin `expand_mul` isn't working for this case

```
>>> expand_mul((A.x-A.y+A.z).express(N))
Attribute Error

>>> (A.x-A.y+A.z).express(N)
(-4*cos(J_theta(t))/3 - 1/3)*N.x + (-2*sqrt(3)*sin(J_theta(t))/3 - 2*cos(J_theta(t))/3 + 1/3)*N.y + (-2*sqrt(3)*sin(J_theta(t))/3 + 2*cos(J_theta(t))/3 - 1/3)*N.z

>>> (A.x-A.y+A.z).express(N).simplify()
(-4*cos(theta)/3 - 1/3)*N.x + (1/3 - 4*sin(theta + pi/6)/3)*N.y + (4*cos(theta + pi/3)/3 - 1/3)*N.z
```

What will be the alternative in this case?

Reply:
I think for vectors we would need to use `simplify` because it would call simplify on it's args which can't be done directly.

Reply:
You can always do `vector.to_matrix(N)` to give you a base sympy type that all functions should work with.

Reply:
> You can always do `vector.to_matrix(N)` to give you a base sympy type that all functions should work with.

It isn't simplifying in desired form, matrix is also making result complicated.

Reply:
What is the full code (with imports) needed to make that expression?

How does `simplify` turn `J_theta(t)` into `theta`? The first expression depends on `t` but after simplifying it does not.

Reply:
theta = dynamicsymbols('J_theta')

Took the 1st one from output but other one from assertion, but they are same.

Reply:
What is the full code (with imports) needed to make that expression?

Reply:
```

> What is the full code (with imports) needed to make that expression?

Using the `Joints` class which is being implemented in this PR or without it?

Reply:
With the PR